src/commands/deploy/deploy.ts

```diff
@@ -51,6 +55,15 @@ export async function deploy(context: IActionContext, target?: vscode.Uri | Site
     await setPreDeployTaskForDotnet(deployContext);
     const { node, isNewWebApp }: IDeployNode = await getDeployNode(deployContext, target, isTargetNewWebApp);
 
+    if (node instanceof TrialAppTreeItem) {
+        const commit: boolean = deployContext.workspace.name === node.metadata.siteName;
```

Comment:
Since it's kind of a niche case, I think it's still worth separating this out as its own function, but leaving it in the deploy folder.